### ISSUE TITLE
A generic type parameter in Kdoc should not be reported

### ISSUE THREAD

**Issue by polarene** [saveourtool/diktat#1929]
## Describe the bug
When documenting a type parameter in Kdoc, the only correct way to do it is using the `@param` tag, for example:
```kotlin
/**
 * @param T the target type to return
 */
fun <T> genericFunc(): T
```
Unfortunately Diktat reports:

> [KDOC_WITHOUT_PARAM_TAG] all methods which take arguments should have @param tags in KDoc: T param isn't present in argument list

Which I believe is a wrong behavior, given that it's taking into consideration the formal parameters only.

## Expected behavior
Diktat should not report a violation for the above example.

## Observed behavior
Diktat reports a KDOC_WITHOUT_PARAM_TAG

## Environment information
* diktat version: 2.0.0
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): plugin
* kotlin version: 1.9.23
* operating system: MacOS 14.3.1

**Comment by orchestr7** [comment]
Sounds very reasonable, but looks to be fixed @DrAlexD please check

**Comment by orchestr7** [comment]
https://github.com/saveourtool/diktat/pull/1873/files ?

**Comment by polarene** [comment]
Thanks for getting back to this bug. The PR you cited seems to be resolving a different problem, that is parameter names enclosed by backticks, not type parameters.
